faunadb-java-dsl/src/main/java/com/faunadb/client/types/Types.java

```diff
@@ -0,0 +1,234 @@
+package com.faunadb.client.types;
+
+import com.faunadb.client.errors.FaunaException;
+import org.joda.time.Instant;
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static java.lang.String.format;
+
+/**
+ * Auxiliary class used to create types representation while decoding user defined objects.
```

Comment:
maybe

"Helper class which contains convenience methods for creating types used to decode user-defined classes" ?